libparsec/crates/client/src/workspace/merge.rs

```diff
@@ -232,21 +269,51 @@ pub(super) fn merge_local_folder_manifest(
     //   that would be considered as bug :/
     // - the fixtures and server data binder system used in the tests
     //   makes it much more likely
-    if local_from_remote.base.author == local_author && !local_speculative {
-        let mut new_local = local.to_owned();
-        new_local.base = local_from_remote.base;
-        return MergeLocalFolderManifestOutcome::Merged(Arc::new(new_local));
+    if remote.author == local_author && !local_speculative {
+        // We should end up with a new local manifest that is strictly equivalent
+        // to the previous one, except that it is now based on the new remote.
+
+        let LocalFolderManifest {
+            base: _, // New remote already set
+            // All the other fields should be overwritten occording to previous local manifest
+            parent: merged_parent,
+            need_sync: merged_need_sync,
+            updated: merged_updated,
+            children: merged_children,
+            local_confinement_points: merged_local_confinement_points,
+            remote_confinement_points: merged_remote_confinement_points,
+            speculative: merged_speculative,
+        } = &mut merge_in_progress;
+
+        *merged_speculative = false;
+        *merged_parent = *local_parent;
+        *merged_need_sync = *local_need_sync;
+        *merged_updated = local.updated;
+        *merged_children = local.children.to_owned();
+        *merged_local_confinement_points = local.local_confinement_points.to_owned();
+        *merged_remote_confinement_points = local.remote_confinement_points.to_owned();
```

Comment:
then `merge_in_process.local_confinement_points` should also be used ?
(I guess we really don't want to have local and remote confinement points out of sync)

Reply:
This is the special case I was explaining in an earlier conversation in slack, and maybe it should be dealt with with a bit more care. But the main idea is: 
- `local_confinement_points` relates to the entries in `children`, so we should modify neither of them in this case.
- `remote_confinement_points` relates to the entries in `base.children`, so we should modify both in this case.

But again, this should be validated with more tests since this was most probably already broken in v2.